src/test/resources/org/rust/ide/inspections/fixtures/unnecessary_parenthesis.rs

```diff
@@ -0,0 +1,9 @@
+fn main() {
+    let x = 5;
+
+    if <warning descr="Unnecessary parenthesis">(x == 5)</warning> {
+        println!("x is five!");
+    } else {
+        println!("x is not five :(");
+    }
+}
```

Comment:
I don't think that we need such tests: it only serves to demonstrate a flaw in a text based implementation and is unlikely to ever fail with PSI based approach. And @zjhmale is right that
`if (true) or (false) {}` is not a good test here.